### PR TITLE
MM-794: returning proper JSON response from APIs

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
#### Summary
returning proper JSON response from APIs

#### Screenshots
Screenshot from 2024-09-03 17-13-26
Screenshot from 2024-09-03 17-14-56

### CODE REVIEW

**Review by raghavaggarwal2308 — COMMENTED** [review]
Please test the PR for error responses of all the APIs also add logging wherever necessary

File: server/plugin/api.go [hunk 1]
```diff
@@ -184,7 +184,7 @@ func (p *Plugin) checkConfigured(next http.Handler) http.Handler {
 		config := p.getConfiguration()
 
 		if err := config.IsValid(); err != nil {
-			http.Error(w, "This plugin is not configured.", http.StatusNotImplemented)
+			p.writeAPIError(w, &APIErrorResponse{Message: "this plugin is not configured", StatusCode: http.StatusNotImplemented})
```
Comment (raghavaggarwal2308) [inline server/plugin/api.go:188]: Why not adding log here?

Comment (raghavaggarwal2308) [reply server/plugin/api.go:188]: Also at other places

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: server/plugin/api.go (hunk 1 above)
Comment (Kshitij-Katiyar) [reply server/plugin/api.go:188]: Added

File: server/plugin/api.go [hunk 2]
```diff
@@ -1439,6 +1442,7 @@ func (p *Plugin) getConfig(w http.ResponseWriter, r *http.Request) {
 func (p *Plugin) getToken(w http.ResponseWriter, r *http.Request) {
 	userID := r.FormValue("userID")
 	if userID == "" {
+		p.client.Log.Error("UserID not found.")
```
Comment (raghavaggarwal2308) [inline server/plugin/api.go:1445]: Add userID

**Review by raghavaggarwal2308 — APPROVED** [review]

File: server/plugin/api.go (hunk 2 above)
Comment (raghavaggarwal2308) [reply server/plugin/api.go:1445]: Oh I see it is empty

**Comment by Kshitij-Katiyar** [comment]
Created a PR on mattermost remote with all these changes hence closing this one

**Pull request closed without merging**